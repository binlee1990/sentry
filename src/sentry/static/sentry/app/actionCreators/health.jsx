const BASE_URL = org => `/organizations/${org.slug}/health/`;

export const doHealthRequest = (
  api,
  {organization, projects, tag, environments, period, timeseries, includePrevious, topk}
) => {
  if (!api) return Promise.reject(new Error('API client not available'));

  const path = timeseries ? 'graph/' : 'top/';
  const query = {
    tag,
    includePrevious,
    statsPeriod: period,
    project: projects,
    ...(topk ? {topk} : {}),
  };

  return api.requestPromise(`${BASE_URL(organization)}${path}`, {
    query,
  });
};
