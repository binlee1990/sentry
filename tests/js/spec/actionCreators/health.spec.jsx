import {Client} from 'app/api';
import {doHealthRequest} from 'app/actionCreators/health';

describe('Health ActionCreator', function() {
  const api = new Client();
  const organization = TestStubs.Organization();
  const project = TestStubs.Project();
  let mock;

  it('requests timeseries', function() {
    mock = MockApiClient.addMockResponse({
      url: '/organizations/org-slug/health/graph/',
    });
    doHealthRequest(api, {
      timeseries: true,
      organization,
      projects: [project],
      environments: [],
      tag: 'release',
      topk: 5,
    });

    expect(mock).toHaveBeenCalled();

    expect(mock.mock.calls[0][1].query).toMatchSnapshot();
  });

  it('requests top', function() {
    mock = MockApiClient.addMockResponse({
      url: '/organizations/org-slug/health/top/',
    });
    doHealthRequest(api, {
      timeseries: false,
      organization,
      projects: [project],
      environments: [],
      tag: 'release',
      topk: 5,
    });

    expect(mock).toHaveBeenCalled();

    expect(mock.mock.calls[0][1].query).toMatchSnapshot();
  });
});
