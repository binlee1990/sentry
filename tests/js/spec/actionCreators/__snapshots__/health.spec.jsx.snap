// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Health ActionCreator requests timeseries 1`] = `
Object {
  "includePrevious": undefined,
  "project": Array [
    Object {
      "hasAccess": true,
      "id": "2",
      "isBookmarked": false,
      "isMember": true,
      "name": "Project Name",
      "slug": "project-slug",
      "teams": Array [],
    },
  ],
  "statsPeriod": undefined,
  "tag": "release",
  "topk": 5,
}
`;

exports[`Health ActionCreator requests top 1`] = `
Object {
  "includePrevious": undefined,
  "project": Array [
    Object {
      "hasAccess": true,
      "id": "2",
      "isBookmarked": false,
      "isMember": true,
      "name": "Project Name",
      "slug": "project-slug",
      "teams": Array [],
    },
  ],
  "statsPeriod": undefined,
  "tag": "release",
  "topk": 5,
}
`;
